src/sidebar/components/tag-editor.js

```diff
@@ -192,47 +193,88 @@ function TagEditor({ onEditTags, tags: tagsService, tagList }) {
   };
 
   /**
-   * Keydown handler for keyboard navigation of the suggestions list
-   * and when the user presses "Enter" or ","" to add a new typed item not
-   * found in the suggestions list
+   * Keydown handler for keyboard navigation of the tag editor field and the
+   * suggested-tags list.
    *
    * @param {KeyboardEvent} e
    */
   const handleKeyDown = e => {
     switch (normalizeKeyName(e.key)) {
       case 'ArrowUp':
+        // Select the previous item in the suggestion list
         changeSelectedItem(-1);
         e.preventDefault();
         break;
       case 'ArrowDown':
+        // Select the next item in the suggestion list
         changeSelectedItem(1);
         e.preventDefault();
         break;
+      case 'Escape':
+        // Clear any entered text, but retain focus
+        inputEl.current.value = '';
+        e.preventDefault();
+        break;
       case 'Enter':
       case ',':
+        // Commit a tag
         if (activeItem === -1) {
           // nothing selected, just add the typed text
           addTag(/** @type {HTMLInputElement} */ (inputEl.current).value);
         } else {
+          // Add the selected tag
           addTag(suggestions[activeItem]);
         }
         e.preventDefault();
         break;
       case 'Tab':
+        // Commit a tag, or tab out of the field if it is empty (default browser
+        // behavior)
+        if (inputEl.current.value === '') {
```

Comment:
Do you agree we should trim() this value?
if there is some whitespace, then this does condition does not work and I dont think it makes sense to lock the focus because add a tab won;'t work either with only white space.